src/OrchardCore/OrchardCore/Localization/NullStringLocalizerFactory.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Extensions.Localization;
+
+namespace OrchardCore.Localization
+{
+    internal class NullStringLocalizerFactory : IStringLocalizerFactory
+    {
+        public IStringLocalizer Create(Type resourceSource) => NullStringLocalizer.Instance;
+        public IStringLocalizer Create(string baseName, string location) => NullStringLocalizer.Instance;
+
+        private class NullStringLocalizer : IStringLocalizer
```

Comment:
If I'm not wrong if anyone asking DI for the current `IStringLocalizer` an exception will be thrown because the `NullStringLocalizer` is `private`

Reply:
You are wrong ;)

Reply:
So can you tell me the expected type?

Reply:
We always inject a typed localizer which resolves `StringLocalizer<TResourceSource>` which injects and uses `IStringLocalizerFactory` which here is our `NullStringLocalizerFactory`.

![Localizer](https://user-images.githubusercontent.com/8586360/54498328-67b41f80-4906-11e9-9098-599cbb312b0f.png)

Reply:
Got it .. it will return `NullStringLocalizerFactory.NullStringLocalizer` as you mentioned because it is defined as nested type

thanks

Reply:
No problem, it was not clear to me either ;)